optuna/visualization.py

```diff
@@ -92,6 +93,74 @@ def _get_intermediate_plot(study):
     return figure
 
 
+def plot_optimization_history(study):
+    # type: (Study) -> None
+    """Inside Jupyter notebook, plot optimization history of all trials in a study.
+
+    Example:
+
+        The following code snippet shows how to plot optimization history inside Jupyter Notebook.
+
+        .. code::
+
+            import optuna
+
+            def objective(trial):
+                ...
+
+            study = optuna.create_study()
+            study.optimize(objective ,n_trials=100)
+
+            optuna.visualization.plot_optimization_history(study)
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+    """
+
+    _check_plotly_availability()
+    init_notebook_mode(connected=True)
+    figure = _get_optimization_history_plot(study)
+    figure.show()
+
+
+def _get_optimization_history_plot(study):
+    # type: (Study) -> Figure
+
+    layout = go.Layout(
+        title='Optimization History Plot',
+        xaxis={'title': 'Trial'},
+        yaxis={'title': 'Objective Value'},
+    )
+
+    trials = [t for t in study.trials if t.state == TrialState.COMPLETE]
+
+    best_values = [float('inf')] if study.direction == StudyDirection.MINIMIZE else [-float('inf')]
+    for trial in trials:
+        if isinstance(trial.value, int):
+            trial_value = float(trial.value)
+        elif isinstance(trial.value, float):
+            trial_value = trial.value
+        else:
+            logger.warning('Your study has a non-numeric value.')
```

Comment:
Since `trial.value` is supposed to be a float as long as the trial is `COMPLETE`, please raise an error instead of warning.